### ISSUE TITLE
Details cost to include "based", "calculated" and "mark up"

### ISSUE THREAD

**Issue by dlabrecq** [project-koku/koku#1397]
As a user, I want the details view to include cost information about "based", "calculated" and "mark up" are not shown. 

This would replace the infrastructure cost and derived costs?

**Comment by dlabrecq** [comment]
//cc @nlcwong @chargio

**Comment by chargio** [comment]
@dayleparker Wouldn't it be "raw", "calculated" and "mark up". And then "calculated" should be split into "infra" and "not infra" in the future:
- Today: raw == cloud
- "Calculated" == using the price list of the cost model
- "Markup" == markup

What do you think?

**Comment by dayleparker** [comment]
@chargio I agree.

Copying the definitions from the Cost Lexicon below and then my own summary.

**Raw costs:** (we added metrics to make the definition broader)
Raw metrics and costs are the metrics and costs provided by a third party that have not been modified in any way, such as the AWS costs provided by the AWS Cost and Usage Report files or the API, and the OpenShift metrics provided through Operator Metering.

Note: Raw costs come from AWS (at the moment anyway) and do not get a price list applied in Cost Management as Amazon applies its own pricing.

**Markup**
The markup is the ratio, positive or negative, between the base cost and the calculated cost.

**Calculated costs**
Calculated costs are the costs stored in the application after a markup is added to the base costs. (This is what gets distributed across resources/parts of the infrastructure)

@sergio, +1 to calculated costs being split into "infra" and "not infra" in the future. I'll attach our diagram to the PR for reference.

also:
**Base costs**
Base costs originate from different sources, such as metrics, inventory, or reports, which are normalized so a markup can be applied. Base costs can be calculated directly in the Cost Management application by applying a price list to metrics or inventory, or they can be calculated externally (i.e. AWS provided costs).  In the Cost Management application’s cost model, a markup is applied to base costs to get calculated costs. 

-------
In summary:
Raw costs (already have prices from AWS) ==> base costs + markup = calculated costs

Metrics/inventory x price list = base costs + markup = calculated costs

-------
[Our previous definitions, no longer using]
Derived cost was previously described as:
"The cost based on the user’s defined price list. For example, this could be calculated as  infrastructure_cost + markup, or on a premises without infrastructure, calculated as CPU usage * rate."

and Infrastructure cost (same as what is now raw costs I think):
"The cost based on the underlying infrastructure. For example, the raw usage data included in an AWS Cost and Usage report."

-------
Does this make sense, @dlabrecq @nlcwong ?

**Comment by dayleparker** [comment]
<img width="535" alt="Screen Shot 2019-12-03 at 14 00 01" src="https://user-images.githubusercontent.com/20275317/70053250-49e23400-15d5-11ea-8b50-a6fba58f3681.png">

**Comment by nlcwong** [comment]
@chargio can you help us map these terminology to what we are showing in the UI (e.g. derived and infrastructure costs in the OpenShift details page?)

**Comment by chargio** [comment]
I believe this can be done:
- Rename infrastructure costs to raw costs (once we get node chargeback they no longer reflect truyly the infra costs)
- Rename derived costs to calculated costs and add there only the price lists calucations
- Add a column for markup
- Keep the total as raw+calculated+markup

@vestival @dayleparker do you agree?

**Comment by dayleparker** [comment]
@chargio I agree. The only thing I'm unclear about is for the total, are you referring to the column currently called "Cost"?  

And for the big total cost number (top right corner of the screen -- will paste screenshot below), this is the total of all calculated costs, before we distribute them across projects, right?

Sorry for the silly question, but where do we show the Distributed Costs?

**Comment by dayleparker** [comment]
Note, with the terminology change, we will need to fix the text inside the Info box in this screen.
<img width="1217" alt="Screenshot 2019-12-04 at 11 25 21" src="https://user-images.githubusercontent.com/20275317/70135265-01388280-168a-11ea-903d-4a5c9c5d09ab.png">

**Comment by chargio** [comment]
Yes, that cost column is the total cost (would it make sense to call it out as such?)
Distributed costs are the costs that we show for projects, clusters, nodes, tags, etc.

**Comment by vestival** [comment]
@chargio I agree and makes sense

**Comment by chargio** [comment]
I've created a new graphic with the proposal.
![Cost Model - New price model](https://user-images.githubusercontent.com/7500590/70521163-0b9dc500-1b3f-11ea-8cfb-8d486e1ffa0a.png)

**Comment by chargio** [comment]
@dayleparker Please comment here and in the lucidchart diagram.

Changes:
- Raw costs are costs coming from a third party (i.e. AWS, Azure, or subs mgmt).
- Calculated costs are those using a price list. The customer should be capable of stating that some of the costs are infra costs (i.e. monthly charges)
- Markups are associated differently, they apply only to the Raw costs and are added to the rest

Then we have another way of looking at this:
- Infrastructure costs are those identified as such (right now they are eq to raw costs but we should differentiate them in the future, because clouds also offer development services that are not infra). 
- Non-infra costs are everything else.

**Comment by dayleparker** [comment]
@chargio  So sorry to have left this hanging!

Looks good to me, and I think by removing 'base costs', it's all simpler to understand if customers don't need that concept.

Am I understanding correctly that customers would only add a markup to raw costs/cloud infrastructure costs (and not the OCP inventory/metrics) to account for the cost of managing the cloud?

Re: the diagram layout -- in the Total Costs box, do we need to call out here that there are infra and non-infra costs? it appears a bit like Infrastructure costs relate only to Inventory & Metrics, and Non-Infra relates only to the Cloud Metrics. Do we need the white boxes for Infra & Non-infra to make the graphic understandable at this time, or maybe:

1. Could we label that box "Total costs (infrastructure and non-infrastructure costs)" and remove the white shapes
2. Could we label that box only "Total costs" and define Total costs in either/both the docs, and maybe in a popover in the UI --> ie. in the "Total cost" column in the OpenShift details page
3. We could realign the white shapes so they don't look to only go with the right or left components in the diagram

Last very tiny edit -- I think the diagram should have an arrowhead on the line from Markup to Total costs for consistency.

What do you think? Maybe I'm missing the significance of the infra/non-infra distinction.

@all - I will edit the terminology definitions, line them up with the table headers, and paste them in the next comment...

**Comment by chargio** [comment]
That is what I want, as the customer can always change the price list with the same result (i.e. you have 1000 and you change it to 1100 in another version)

I will update the graphic following on your comments and upload it again

**Comment by dayleparker** [comment]
OpenShift details columns:

Project names   |    Month over month change  |  Raw cost   |   Markup   | Calculated cost  |  Total cost

Previously:
(Raw cost was 'Infrastructure cost')
(Calculated cost was 'Derived cost')
(Total cost was 'Cost')
---
Text for popovers/hover text (if desired):

**Raw cost**
Costs provided by a third party that have not been modified in any way.

(or: "a third-party cloud source"? "cloud infrastructure sources"? Or: "Cloud infrastructure costs provided by a third party that have not been modified in any way." ?)

**Markup**
A positive or negative ratio modifying the raw cloud costs to account for the additional costs of managing the cloud.

**Calculated cost**
Costs stored in the application after a custom price list is applied to inventory and metrics.

(or ... 'applied to OpenShift inventory and metrics'?) 

**Total cost**
All infrastructure and non-infrastructure costs, distributed to the appropriate resource using tags.  Calculated as (raw costs + markup) + calculated costs.

One last question, @chargio :
Re: Total costs --> in the diagram, we have the "Total costs" label before the costs are distributed. But in the app, I think the Total cost column in OpenShift detailsshows the costs distributed. Should we modify the diagram in some way to reflect that?

**Comment by chargio** [comment]
Calculated costs: the definition is right. If we extend our price list to on-premises it would still be valid.

The total costs is shown in the dashboard too, so I believe we should leave it as it is. We are just distributing raw, markup and calculated costs into business elements.

**Comment by nbon12** [comment]
Sprint Planning Meeting Notes:
Andrew: "With the API returning things like derived cost, do we need to change the API to return different words"?
Natalie: "We need another meeting with Sergio and Dayle to figure out what is wanted to show, and what does it mean, and what are we going to call it."

**Comment by chargio** [comment]
I am happy to have the conversation, please send an invite to me and @dayleparker 

Meanwhile it would be great to know what is missing in the current slidedeck so we can go through it.

**Comment by chambridge** [comment]
This issue drove the discussion but the resulting work will be tied to the following issue: https://github.com/project-koku/koku/issues/1762
